// form-schemas/register.js
var validator =  require('validator');
var customValidator = require('custom-form-validator');

var registerSchema = [
    {
        name: 'email',
        label: 'Email',
        type: 'text',
        value: '',
        hasError: false,
        validators: [{fn: validator.isEmail, message: 'Should be an email'}]
    },
    {
        name: 'password',
        label: 'Password',
        type: 'text',
        value: '',
        options: {type: 'password'},
        hasError: false,
        validators: [
            {
                fn: customValidator.isValidPassword,
                message: 'At least 4 characters'
            }
        ]
    },
    {
        name: 'password2',
        label: 'Retype the password',
        type: 'text',
        value: '',
        options: {type: 'password'},
        hasError: false,
        validators: [
            {
                fn: customValidator.isValidPassword,
                message: 'At least 4 characters'
            }
        ]
    }
];

module.exports = registerSchema;
